activerecord/lib/active_record/migration.rb

```diff
@@ -1203,7 +1205,10 @@ def initialize(direction, migrations, schema_migration, target_version = nil)
       validate(@migrations)
 
       @schema_migration.create_table
-      ActiveRecord::InternalMetadata.create_table
+
+      if ActiveRecord::Base.connection.supports_metadata_table?
```

Comment:
It will be really easy to forgot to add this check later. Any chance to move this check to `InternalMetadata` class itself?

Reply:
Moved most of the conditional logic inside the InternalMetadata class